vault/identity_store_groups.go

```diff
@@ -219,6 +219,8 @@ func (i *IdentityStore) handleGroupUpdateCommon(ctx context.Context, req *logica
 		switch {
 		case groupByName == nil:
 			// Allowed
+		case newGroup && groupByName != nil:
+			return logical.ErrorResponse("new group name is already in use"), nil
 		case group.ID == "":
 			group = groupByName
```

Comment:
Using the non-cloned `groupByName` result will always lead to the bug. You need to either clone the `groupByName` as I did in https://github.com/hashicorp/vault/pull/10582 or you need to fail on that case.

The case you added essentially makes the `group.ID == ""` case dead code which solves the bug as well because the only way `group.ID == ""` can currently occur is when it is a `newGroup`. However that case should be removed entirely.

Reply:
For me the bug is not that the update is not happening but that it should return an error, the documentation says it needs an ID to do the update.

Reply:
Yep, agreed, I listed this as an alternate in my PR. But the code path for `group.ID == ""` is wrong and should be deleted. The case you added will always trigger instead of `group.ID == ""`, so that case is now dead code. It should not be left in. If, in the future, it was triggered again somehow, then this bug would re-appear.